install.ps1

```diff
@@ -8,6 +8,15 @@ Write-Host -Object "https://github.com/nimsandu/spicetify-bloom"
 # Give time for user to cancel via CTRL+C
 Start-Sleep -Seconds 3
 
+# Check if Spotify installed
+if ( -not (Test-Path -Path "$env:APPDATA/Spotify") ) {
+  if ( -not (Get-AppxPackage | Where-Object -Property Name -Match "^SpotifyAB") ) {
+    Write-Host -Object "Spotify not installed!" -ForegroundColor Red
+    Start-Sleep -Seconds 3
+    exit

```

Comment:
Why don't you make it install Spotify instead of exiting?

Reply:
Not really, just running Spotify once suffices Spicetify. This way is easy to implement: You install Spotify, wait for 20 seconds, run Spotify if it's not running, then kill it.

Reply:
![image](https://user-images.githubusercontent.com/115353812/220351547-8fd6f7fb-9545-4228-9007-532b0ec3c4e3.png)
proof for 30 seconds requirement

I can implement tracking uptime but hiding spotify client window requires a lot of code:

<details><summary>Hide Spotify window</summary>
<p>

```powershell
function Push-Window {
	[CmdletBinding()]
	param ([IntPtr]$Handle, [String]$ProcessName)

	[IntPtr[]]$Handles = @($Handle)
	if (!$ProcessName) { $ProcessName = "--Unknown--" }

	if (!$Windows.ContainsKey($ProcessName)) {
		$windows.Add($ProcessName, (New-Object $intPtrList))
	}
	if ($Handle -eq [IntPtr]::Zero) {
		[IntPtr[]]$Handles = @(Get-Process $ProcessName | ForEach-Object { $_.MainWindowHandle } )
	}
	$windows[$ProcessName].AddRange($Handles)
}

function Hide-Window {
	[CmdletBinding()]
	param (
		[Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
		[Alias("MainWindowHandle", "Handle")]
		[IntPtr]$WindowHandle = [IntPtr]::Zero,

		[Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
		[Alias("ProcessName")]
		[String]$Name
	)
	process {
		(Push-Window $WindowHandle $Name) | ForEach-Object { [Tricks]::HideWindow( $_ ) }
	}
}

Add-Type @"
using System;
using System.Runtime.InteropServices;
public class Tricks {
   [DllImport("user32.dll")]
   private static extern bool ShowWindowAsync(IntPtr hWnd, int nCmdShow);
	public static void ShowWindow(IntPtr hWnd) { ShowWindowAsync(hWnd,5); }
	public static void HideWindow(IntPtr hWnd) { ShowWindowAsync(hWnd,0); }
}
"@

$dictionary = [System.Collections.Generic.Dictionary``2]
$list = [System.Collections.Generic.List``1]
$string = [String]
$intPtr = [IntPtr]

$intPtrList = $list.AssemblyQualifiedName -replace "1,", "1[[$($intPtr.AssemblyQualifiedName)]],"
$windows = New-Object ($dictionary.AssemblyQualifiedName -replace "2,", "2[[$($string.AssemblyQualifiedName)],[$intPtrList]],")

Wait-Process spotify | Hide-Window
```

</p>
</details>

Reply:
Don't hide it, just **kill** it. Literally `taskkill -f -im spotify.exe`.

Reply:
As I said Spotify should work for 30 seconds

Reply:
Hiding spotify window is required because an average user likely to close suddenly apperead window it at once